hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/InvokeOnMembers.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.scheduledexecutor.impl;
+
+import com.hazelcast.core.Member;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.OperationFactory;
+import com.hazelcast.spi.OperationService;
+import com.hazelcast.spi.annotation.PrivateApi;
+import com.hazelcast.spi.serialization.SerializationService;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+/**
+ * Executes an operation on a set of targets.
+ * Similar to {@link com.hazelcast.spi.impl.operationservice.impl.InvokeOnPartitions} but for members.
+ */
+@PrivateApi
+public final class InvokeOnMembers {
```

Comment:
Some duplication with the `InvokeOnPartitions`. Perhaps we can move this to `com.hazelcast.spi.impl.operationservice.impl` and have a common base class